### PR TITLE
Add types of nodes, and node implementations.

### PR DESCRIPTION
**Pull request by johnnymatthews**
This is a _prettifying_ of a [HackMD](https://hackmd.io/j1PQOmB3ReKEoMoQuN-tEg?view) put together by internal PL IPFS engineers and dev-rel folks.

It's pretty advanced as it stands right now, and could do with some simplifying for a more general reader. However, right now it serves a purpose.

### CODE REVIEW

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/concepts/nodes.md`:
- The following link returned HTTP status code 404 (Not Found):
  https://github.com/ipfs/js-ipfs/blob/master/packages/ipfs-core/src/runtime/config-nodejs.js
- ‘with the help of’ might be wordy. Consider a shorter alternative.
  ```
  ...connect to a browser node using WebRTC, with the help of a centralized ws-we...
                                             ^^^^^^^^^^^^^^^^
  ```
- To make your text as clear as possible to all readers, do not use this foreign term, except as a technical term in engineering. Possible alternatives are “list of additions”, “additional information”, “appendix”, “supplement”.
  ```
  ...reload Relay Bootstrap Delegate routing Addenda # Preload When users want to...
                                             ^^^^^^^
  ```
- Possible missing comma found.
  ```
  ...(opens new window) as regular bootstrap nodes but have the string 'preload' ...
                                             ^^^^^
  ```
- Possible missing comma found.
  ```
  ...de. Publishing works with arbitrary CID codecs as the js-delegate-content mo...
                                             ^^^^^^
  ```
- Possible missing comma found.
  ```
  ...(opens new window) as regular bootstrap nodes but have the string 'preload' ...
                                             ^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/concepts/nodes.md`:
- The following link returned HTTP status code 404 (Not Found):
  https://github.com/ipfs/js-ipfs/blob/master/packages/ipfs-core/src/runtime/config-nodejs.js

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
- Check of `docs/concepts/nodes.md` came back clean!

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
- Check of `docs/concepts/nodes.md` came back clean!

File: docs/concepts/nodes.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+---
+title: "Nodes"
+description: "Computers running IPFS are called nodes. Nodes are the most imporant aspect of IPFS - without nodes running the IPFS daemon, there would be no such thing as IPFS. This page discusses what nodes are, current IPFS implementations, and the types of services different nodes can offer."
```
Comment (momack2) [inline docs/concepts/nodes.md:3]:
```suggestion
description: "Participants in the IPFS network are called nodes. Nodes are the most important aspect of IPFS - without nodes running the IPFS daemon, there would be no  IPFS network. This page discusses what nodes are, current IPFS implementations, and the types of services different nodes can offer."
```

File: docs/concepts/nodes.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+---
+
+# Nodes
+
+Computers running IPFS are called _nodes_. Nodes are the most crucial aspect of IPFS - without nodes running the IPFS daemon, there would be no such thing as IPFS in the first place.
```
Comment (momack2) [inline docs/concepts/nodes.md:8]:
```suggestion
Participants in the IPFS network are called _nodes_. Nodes are the most crucial aspect of IPFS - without nodes running the IPFS daemon, there would be no IPFS Network.
```

File: docs/concepts/nodes.md (hunk 1 above)
Comment (momack2) [reply docs/concepts/nodes.md:3]: there *would* be such a thing as IPFS - the IPFS protocol - even if no nodes were running it. A protocol doesnt require running implementations in order to be valid. However the network does require active nodes - so rephrased

**Review by momack2 — APPROVED** [review]

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/concepts/nodes.md`:
- Possible typo detected.
  ```
  .... JS-ipfs nodes remain connected to the libp2p swarm ports of all preload no...
                                             ^^^^^^
  ```
- Possible typo detected.
  ```
  ... bootstrap nodes, so they will maintain libp2p swarm connections to them at ...
                                             ^^^^^^
  ```
- The preposition ‘with’ is not usually used with the verb ‘talk’. Try using a more formal alternative, or replace the preposition.
  ```
  ...e with WebSockets/webRTC transports can talk with a Go-IPFS node that only c...
                                             ^^^^^^^^^
  ```
- Possible typo detected.
  ```
  ... bootstrap nodes, so they will maintain libp2p swarm connections to them at ...
                                             ^^^^^^
  ```

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/concepts/nodes.md`:
- Possible typo detected.
  ```
  .... JS-ipfs nodes remain connected to the libp2p swarm ports of all preload no...
                                             ^^^^^^
  ```
- Possible typo detected.
  ```
  ... bootstrap nodes, so they will maintain libp2p swarm connections to them at ...
                                             ^^^^^^
  ```
- The preposition ‘with’ is not usually used with the verb ‘talk’. Try using a more formal alternative, or replace the preposition.
  ```
  ...e with WebSockets/webRTC transports can talk with a Go-IPFS node that only c...
                                             ^^^^^^^^^
  ```
- Possible typo detected.
  ```
  ... bootstrap nodes, so they will maintain libp2p swarm connections to them at ...
                                             ^^^^^^
  ```

**Pull request merged**